staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/objectmeta/coerce_test.go

```diff
@@ -161,6 +162,195 @@ func TestMalformedObjectMetaFields(t *testing.T) {
+			returnUnknownFieldPaths: true,
+			expectedErr:             "json: cannot unmarshal number into Go struct field ObjectMeta.annotations of type map[string]string",
+		},
+		{
+			obj:                     unknownAndMalformed,
+			dropMalformedFields:     true,
+			returnUnknownFieldPaths: true,
+			expectedObject: &metav1.ObjectMeta{
+				Name: "my-meta",
+				Labels: map[string]string{
+					"foo": "bar",
+				},
+			},
+			expectedUnknownPaths: []string{"metadata.unknownField"},
+		},
+
+		{
+			obj:                     unknownOnly,
+			dropMalformedFields:     false,
+			returnUnknownFieldPaths: false,
+			expectedObject: &metav1.ObjectMeta{
+				Name: "my-meta",
+				Labels: map[string]string{
+					"foo": "bar",
+				},
+			},
+		},
+		{
+			obj:                     unknownOnly,
+			dropMalformedFields:     true,
+			returnUnknownFieldPaths: false,
+			expectedObject: &metav1.ObjectMeta{
+				Name: "my-meta",
+				Labels: map[string]string{
+					"foo": "bar",
+				},
+			},
+		},
+		{
+			obj:                     unknownOnly,
+			dropMalformedFields:     false,
+			returnUnknownFieldPaths: true,
+			expectedObject: &metav1.ObjectMeta{
+				Name: "my-meta",
+				Labels: map[string]string{
+					"foo": "bar",
+				},
+			},
+			expectedUnknownPaths: []string{"metadata.unknownField"},
+		},
+		{
+			obj:                     unknownOnly,
+			dropMalformedFields:     true,
+			returnUnknownFieldPaths: true,
+			expectedObject: &metav1.ObjectMeta{
+				Name: "my-meta",
+				Labels: map[string]string{
+					"foo": "bar",
+				},
+			},
+			expectedUnknownPaths: []string{"metadata.unknownField"},
+		},
+
+		{
+			obj:                     malformedOnly,
+			dropMalformedFields:     false,
+			returnUnknownFieldPaths: false,
+			expectedErr:             "json: cannot unmarshal number into Go struct field ObjectMeta.annotations of type map[string]string",
+		},
+		{
+			obj:                     malformedOnly,
+			dropMalformedFields:     true,
+			returnUnknownFieldPaths: false,
+			expectedObject: &metav1.ObjectMeta{
+				Name: "my-meta",
+				Labels: map[string]string{
+					"foo": "bar",
+				},
+			},
+		},
+		{
+			obj:                     malformedOnly,
+			dropMalformedFields:     false,
+			returnUnknownFieldPaths: true,
+			expectedErr:             "json: cannot unmarshal number into Go struct field ObjectMeta.annotations of type map[string]string",
+		},
+		{
+			obj:                     malformedOnly,
+			dropMalformedFields:     true,
+			returnUnknownFieldPaths: true,
+			expectedObject: &metav1.ObjectMeta{
+				Name: "my-meta",
+				Labels: map[string]string{
+					"foo": "bar",
+				},
+			},
+		},
+	}
+	for _, tc := range testcases {
+		opts := ObjectMetaOptions{
+			ReturnUnknownFieldPaths: tc.returnUnknownFieldPaths,
+			DropMalformedFields:     tc.dropMalformedFields,
+		}
+		obj, _, unknownPaths, err := GetObjectMetaWithOptions(tc.obj, opts)
+		klog.Warningf("o: %v\nse: %v\ne: %v\n", obj, unknownPaths, err)
```

Comment:
nit: output to t.Log instead of klog (or drop the debug output if we don't need it)?

Reply:
removed